src/core/raster/qgsrastertransparency.cpp

```diff
@@ -97,7 +87,7 @@ int QgsRasterTransparency::alphaValue( double value, int globalTransparency ) co
   //if a match was found use the stored transparency percentage
   if ( myTransparentPixelFound )
   {
-    return static_cast< int >( static_cast< float >( globalTransparency ) * ( 1.0 - ( myTransparentPixel.percentTransparent / 100.0 ) ) );
+    return static_cast< int >( static_cast< float >( globalTransparency ) * myTransparentPixel.opacity );
```

Comment:
a transparency multiplied by an opacity... That gives me some headaches :-)  Is globalTransparency properly named or is it a globalOpacity ?

Reply:
It's actually an alpha value :rofl: (0-255, integer). Let me fix this too!

Reply:
Ok, should be fixed by [729c814](https://github.com/qgis/QGIS/pull/56598/commits/729c81431151efd1e241a3e253a548eaa29fb454)